src/main/java/neko/Neko.java

```diff
@@ -0,0 +1,149 @@
+package neko;
+import neko.task.Task;
+import neko.task.TaskList;
+import neko.ui.Ui;
+
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.io.IOException;
+
+/**
+ * The Neko class represents a cute cat chatbot that serves as a task manager.
+ *  It handles user input and executes commands related to task management, such as adding,
+ *  deleting, marking tasks as done, and listing all tasks. The tasks are stored in a file,
+ *  which is loaded when the Neko object is instantiated.
+ *
+ * @author  Siow Rui Ming
+ * @version 0.1
+ * @since   2024-08-19
+ */
+
+
+public class Neko {
+    private static final String COMMAND_EXIT = "bye";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_MARK = "mark";
+    private static final String COMMAND_UNMARK = "unmark";
+    private static final String COMMAND_ADD = "add";
+    private static final String COMMAND_DELETE = "delete";
+    private static final String COMMAND_FIND = "find";
+    private static final String FILE_PATH = "./data/neko.txt";
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+
+    /**
+     * Constructs a new Neko object with the specified file path to store the tasks.
+     * Initializes the UI, storage, and task list. If the tasks file is not found, it initializes
+     * an empty task list.
+     *
+     * @param filePath The path of the file to store tasks.
+     */
+
+    public Neko(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.loadTasks());
+        } catch (FileNotFoundException e) {
+            ui.showMessage(e.getMessage());
+            tasks = new TaskList(new ArrayList<>());
+        }
+    }
+
+    /**
+     * The main method which makes use of the run method to start the chatbot.
+     *
+     * @param args Unused.
+     */
+
+    public static void main(String[] args) {
+        Neko neko = new Neko(FILE_PATH);
+        neko.run();
+    }
+
+    /**
+     * Runs the Neko chatbot, displaying a greeting, processing user commands,
+     * and continuing until the exit command is received.
+     * Handles user input by calling handleInput method.
+     */
+
+    public void run() {
+
+        ui.showGreeting();
+
+        String input = ui.getUserCommand();
+        while (!input.equals(COMMAND_EXIT)) {
```

Comment:
I like how you used a simple while condition to check for the exit command. Should we also handle the possibility of input being null to avoid potential NullPointerException?